src/app/entity/entity.ts

```diff
@@ -23,25 +23,47 @@
  */
 export class Entity {
 
-  private _id: string;
+  /**
+   * The unique id of this entity.
+   */
+  private readonly entityId: string;
```

Comment:
I've left this to be `entityId` on purpose, simply because it will create more distinguishable datasets in the database as otherwise we would have an `id` and an `_id` field. And it's hidden through the getter and constructor in the entity class so the name doesn't matter anyways.